File: 35_SearchInsertPosition/SearchInsertPosition.java

class Solution {
    public int searchInsert(int[] nums, int target) {
        // own
        int near = 0;
        int val = 0;
        for(int i = 0 ; i < nums.length; i++) {
            if (nums[i] == target) {
                return i;
            }
            else if (nums[i] < target) {
                near = i;
                val = nums[i];
                }
        }
        if(val == 0) return near;
        return near+1;
    }
}

File: 35_SearchInsertPosition/SearchInsertPosition.py

class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        # own
        near = 0
        val = 0
        for i in range(len(nums)):
            if (nums[i] == target):
                return i
            elif (nums[i] < target):
                near = i
                val = nums[i]
        if(val == 0):
            return near
        return near+1
